src/main/java/gyro/google/compute/RegionInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,193 @@
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
+    }
+
+    @Override
+    public void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames)
+        throws Exception {
+        // TODO: target pool resource
+        // https://github.com/perfectsense/gyro-google-provider/issues/79
+        //    private List<AbstractTargetPoolResource> targetPools;
+        boolean shouldPatch = false;
+
+        InstanceGroupManager instanceGroupManager = new InstanceGroupManager();
+
+        for (String changedFieldName : changedFieldNames) {
+            // template changes
+            if (changedFieldName.equals("instance-template")) {
+                setInstanceTemplate();
+            } else if (changedFieldName.equals("target-size")) {
+                instanceGroupManager.setTargetSize(getTargetSize());
+                shouldPatch = true;
+            } else if (changedFieldName.equals("auto-healing-policy")) {
+                List<ComputeInstanceGroupManagerAutoHealingPolicy> diffableAutoHealingPolicy = getAutoHealingPolicy();
+                List<InstanceGroupManagerAutoHealingPolicy> autoHealingPolicies = null;
+
+                if (diffableAutoHealingPolicy.isEmpty()) {
+                    autoHealingPolicies = Data.nullOf(List.class);
+                } else {
+                    autoHealingPolicies = diffableAutoHealingPolicy
+                        .stream()
+                        .map(ComputeInstanceGroupManagerAutoHealingPolicy::copyTo)
+                        .collect(Collectors.toList());
+                }
+                instanceGroupManager.setAutoHealingPolicies(autoHealingPolicies);
+                shouldPatch = true;
+            } else if (changedFieldName.equals("named-port")) {
+                // TODO: investigate as GCP UI allows updating of named ports.
+            }
+        }
+
+        if (shouldPatch) {
+            patch(instanceGroupManager);
+        }
+    }
+
+    @Override
+    public void doDelete(GyroUI ui, State state) throws Exception {
+        Compute client = createClient(Compute.class);
+        Operation operation = client.regionInstanceGroupManagers()
+            .delete(getProjectId(), getRegion(), getName())
+            .execute();
+        waitForCompletion(client, operation);
+    }
+
+    @Override
+    public void copyFrom(InstanceGroupManager model) {
+        super.copyFrom(model);
+
+        setDistributionPolicy(Optional.ofNullable(model.getDistributionPolicy())
+            .map(e -> {
+                ComputeDistributionPolicy computeDistributionPolicy = newSubresource(
+                    ComputeDistributionPolicy.class);
+                computeDistributionPolicy.copyFrom(e);
+                return computeDistributionPolicy;
+            })
+            .orElse(null));
+        // Do NOT update region with a full url as this should be a name.
+        //        setRegion(model.getRegion());
```

Comment:
Shouldn't this be fine now as the `@Immutable` annotation is present ??

Reply:
GCP doesn't accept a full url in the request.